.github/workflows/schema-ci.yml

```diff
@@ -0,0 +1,23 @@
+name: schema-ci
+
+on:
+  pull_request:
+    paths:
+      - "schemas/**"
+    branches: [main]
+
+permissions:
+  contents: read
+  pull-requests: read
```

Comment:
Why do we need this?

Reply:
Let me test around with less permissions and see.

Reply:
`contents: read` is required by checkout action to have access to the repo content. (for private repos - I don't know if it's required for public repos or not)
![image](https://github.com/Azure/azure-dev/assets/64026625/af057991-6c33-4f9c-bbf9-b83721d9ed69)

`pull-requests: read` didn't make a difference in the workflow run so, I will remove it.

Explicitly specifying the workflow permissions is required by Microsoft or it will be disabled on February 1, 2024.
See https://docs.opensource.microsoft.com/github/apps/permission-changes/#what-steps-must-i-take-to-avoid-a-bad-time

Reply:
Thank you for checking.